Zend/zend_builtin_functions.stub.php

```diff
@@ -38,25 +39,28 @@ function is_a($object, string $class_name, bool $allow_string = false): bool {}
 
 function get_class_vars(string $class_name): array|false {}
 
-function get_object_vars(object $obj): array {}
+function get_object_vars(object $object): array {}
 
-function get_mangled_object_vars(object $obj): array {}
+function get_mangled_object_vars(object $object): array {}
 
-function get_class_methods(string|object $class): array {}
+/** string|object $class */
+function get_class_methods($class): array {}
 
+/** @param object|string $object_or_class */
 function method_exists($object_or_class, string $method): bool {}
 
+/** @param object|string $object_or_class */
 function property_exists($object_or_class, string $property_name): bool {}
```

Comment:
It looks like there's some inconsistency on whether or not to include the `_name` suffix. E.g. above we have `string $method` and here we have `string $property_name`. Then below we have `string $class_name` but also `string $extension`.

Reply:
Yes, I had problems with this. Probably I should have written `$property` instead of `$property_name`, so that we don't include the `_name` suffix when it's not needed. I think it's only needed when its absence is ambiguous (I consider `$class_name` this case). Would this be ok for you or do you have a better idea?

Reply:
So the rule I try to follow is:
- Params in connection with classes, objects, and interfaces are called `$class_name`, `$object_name`, `$interface_name` if their only type is string, `$class`, `$object`, `$interface` otherwise
- Params in connection with constants are called `$constant_name` and `$value`. We could go with `$name` if the function names would indicate that they're dealing with constants (e.g. `define()`).
- Params in connection with properties are called `$property` since properties only have string representation

Reply:
> I'm not sure though what to do with the $object_or_class parameter names. Should we call every parameter which both accept an object or a string?

Probably. I think it's more obvious to distinguish `$class` and `$object_or_class`, rather than `$class_name` and `$class`.

Reply:
> Probably. I think it's more obvious to distinguish $class and $object_or_class, rather than $class_name and $class.

I like this idea :) That said, I renamed a few parameter names: now, we either have `$class` or `$object_or_class`. Additionally, I got rid of `$trait_name`, `$interface_name`, and `$function_name` in order to use `$trait`, `$interface`, and `$function`. I hope these are all ok.